causal_testing/json_front/json_class.py

```diff
@@ -197,27 +201,30 @@ def _setup_test(self, causal_test_case: CausalTestCase, estimator: Estimator) ->
         minimal_adjustment_set = self.causal_specification.causal_dag.identification(causal_test_case.base_test_case)
         treatment_var = causal_test_case.treatment_variable
         minimal_adjustment_set = minimal_adjustment_set - {treatment_var}
-        estimation_model = estimator(
-            treatment=treatment_var.name,
-            treatment_value=causal_test_case.treatment_value,
-            control_value=causal_test_case.control_value,
-            adjustment_set=minimal_adjustment_set,
-            outcome=causal_test_case.outcome_variable.name,
-            df=causal_test_engine.scenario_execution_data_df,
-            effect_modifiers=causal_test_case.effect_modifier_configuration,
-        )
-
-        self.add_modelling_assumptions(estimation_model)
+        if "formula" in test:
+            estimation_model = test["estimator"](
+                treatment=treatment_var.name,
+                treatment_value=causal_test_case.treatment_value,
+                control_value=causal_test_case.control_value,
+                adjustment_set=minimal_adjustment_set,
+                outcome=causal_test_case.outcome_variable.name,
+                df=causal_test_engine.scenario_execution_data_df,
+                effect_modifiers=causal_test_case.effect_modifier_configuration,
+                formula=test["formula"],
+            )
+        else:
```

Comment:
I agree it's not nice having the if statement. However I think the problem with the line you provided is that it assumes the estimator has a `formula` attribute in its init function. If there isn't one, it will produce an error even if the formula value is `None`. 

A potentially neater alternative could be to do something like this?

```
init_dict = dict(
            treatment=treatment_var.name,
            treatment_value=causal_test_case.treatment_value,
            control_value=causal_test_case.control_value,
            adjustment_set=minimal_adjustment_set,
            outcome=causal_test_case.outcome_variable.name,
            df=causal_test_engine.scenario_execution_data_df,
            effect_modifiers=causal_test_case.effect_modifier_configuration
        )
if "formula" in test:
            init_dict["formula"] = test["formula"]
        
estimation_model = test["estimator"](**init_dict)
```

Reply:
Ah yeah, that makes sense actually. Fair point.